docs/how-to-guides/performance_analysis.md

```diff
@@ -0,0 +1,206 @@
+# Performance analysis
+
+## Introduction
+
+Autoware is a real-time system, and it is important to have a small response time. If Autoware appears to be slow, it is imperative to conduct performance measurements and implement improvements based on the analysis. However, Autoware is a complex software system comprising numerous ROS 2 nodes, potentially complicating the process of identifying bottlenecks. To address this challenge, we will discuss methods for conducting detailed performance measurements for Autoware and provide case studies. It is worth noting that multiple factors can contribute to poor performance, such as scheduling and memory allocation in the OS layer, but our focus in this page will be on user code bottlenecks. The outline of this section is as follows:
+
+- Performance measurement
+  - Single node execution
+  - Prepare separated cores
+  - Run single node separately
+  - Measurement and visualization
+- Case studies
+  - Sensing component
+  - Planning component
+
+## Performance measurement
+
+Improvement is impossible without precise measurements.
+To measure the performance of the application code, it is essential to eliminate any external influences.
+Such influences include interference from the operating system and CPU frequency fluctuations.
+Scheduling effects also occur when core resources are shared by multiple threads.
+This section outlines a technique for accurately measuring the performance of the application code for a specific node.
+Though this section only discusses the case of Linux on Intel CPUs, similar considerations should be made in other environments.
+
+### Single node execution
+
+To eliminate the influence of scheduling, the node being measured should operate independently, using the same logic as when the entire Autoware system is running.
+To accomplish this, record all input topics of the node to be measured while the whole Autoware system is running.
+To achieve this objective, a tool called [`ros2_single_node_replayer`](https://github.com/sykwer/ros2_single_node_replayer) has been prepared.
```

Comment:
Are the nodes with service/action supported? If not, it might worth noting the limitation.
e.g., "Nodes with ROS service/action are not supported, but may become available in the future"